Update and Move eng/common (#930)
Azure/azure-sdk-tools: eng/common/README.md

2. `azure-sdk-tools - sync - eng-common` is run automatically. It creates **Sync PRs** in each of the connected language repositories using the format `Sync eng/common directory with azure-sdk-tools for PR {Tools PR Number}`. Each **Sync PR** will contain a link back to the **Tools PR** that triggered it.
3. More changes pushed to the **Tools PR**, will automatically triggered new pipeline runs in the respective **Sync PRs**. The **Sync PRs** are used to make sure the changes would not break any of the connected pipelines.
4. Once satisfied with the changes;

>     - First make sure all checks in the **Sync PRs** are green and approved. The **Tools PR** contains links to all the **Sync PRs**.

Reviewer: ```suggestion
    - First make sure all checks in the **Sync PRs** are green and approved. The **Tools PR** contains links to all the **Sync PRs**. If for some reason the PRs is blocked by a CI gate get someone with permission to override and manually merge the PR. 
```

Reviewer: I also think it might be worth adding a note about how to run the script locally to check the status of each of the sync PRs. I already added that support to the script.

>     - Next merge all the **Sync PRs** by approving the Review gate on the `azure-sdk-tools - sync - eng-common` run for the **Tools PR**. You can get to this by clicking on `Details` -> `View more details on Azure Pipelines`. You need EngSys permissions to reach the `azure-sdk-tools - sync - eng-common` pipeline.

Reviewer: ```suggestion
    - Next approve the `VerifyAndMerge` job for the `azure-sdk-tools - sync - eng-common` pipeline triggered by your **Tools PR** which will automatically merge all the **Sync PRs**.  You need `azure-sdk` devops contributor permissions to reach the `azure-sdk-tools - sync - eng-common` pipeline.
```

>     - In the `azure-sdk-tools - sync - eng-common` run for you **Tools PR** click on `Review` -> `Approve`. This will cause all the **Sync PRs** to be merged.

Reviewer: ```suggestion
```

>     - Finally merge the **Tools PR**. Each **Sync PR** contains the link to the corresponding **Tools PR**. 

Reviewer: ```suggestion
    - Finally merge the **Tools PR**. 
```

3. More changes pushed to the **Tools PR**, will automatically triggered new pipeline runs in the respective **Sync PRs**. The **Sync PRs** are used to make sure the changes would not break any of the connected pipelines.
4. Once satisfied with the changes;
    - First merge all the **Sync PRs**. The **Tools PR** contains links to all the **Sync PRs**
    - Finally merge the **Tools PR**. Each **Sync PR** contains the link to the corresponding **Tools PR**. 

> See [detailed eng/common Workflow](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/281/Engineering-Systems-Tools?anchor=eng%2Fcommon-workflow)

Reviewer: While folks may not be able to see the pipeline I think it is still worth including most of the steps in this file.

Reviewer: I don't like that we are adding a protected link here as we will have to add this to our ignore-links.txt file in every repo which I'd like to avoid. Lets start by not having this link and get as much detail as we can directly in the MD file here. If we need more details or images we can add them here later.
